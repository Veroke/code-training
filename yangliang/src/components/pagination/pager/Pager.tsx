import { defineComponent, onMounted } from "vue";
import { addLog } from "../../../log/log";
import './pager.css';

type PagerProps = {
  page: number;
  onClick: (page: number) => void;
  active: boolean;
}

export default defineComponent({
    name: "Pager",
    props: [
      'page',
      'onClick',
      'active'
    ],
    setup(props: PagerProps, { attrs, emit, slots }) {
      onMounted(()=>{
        addLog({
          level: "info",
          message: {
            component: "Pager",
            props,
            msg: 'Some One Use Pager Component'
          }
        })
      })
    
      return () => {
        return (
          <li class = {
            `pager ${props.active ? 'active' : ''}`
          }
          onClick={()=>{
            props.onClick(props.page)
          }}>
              {props.page}
          </li>
        );
      };
    },
});