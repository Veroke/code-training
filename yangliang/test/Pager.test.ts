import { shallowMount } from "@vue/test-utils";
import { expect, test, vi } from "vitest";
import Pager from "../src/components/pagination/pager/Pager";
import Mock from "mockjs";

const pagerTestData = Mock.mock({
  'page:1-100': 100,
  'active|1-2': true,
});

test("mount component", () => {
    const wrapper = shallowMount(Pager, {
      props: {
        page: pagerTestData.page,
        active: pagerTestData.active,
        onClick: vi.fn(),
      },
    });
  
    expect(wrapper.html()).toMatchSnapshot();
});


