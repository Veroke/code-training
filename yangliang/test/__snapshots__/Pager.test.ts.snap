// Vitest Snapshot v1

exports[`active component 1`] = `
"<li class=\\"pager active\\">
  <!---->
</li>"
`;

exports[`mount component 1`] = `"<li class=\\"pager active\\">1</li>"`;

exports[`no-active component 1`] = `
"<li class=\\"pager \\">
  <!---->
</li>"
`;
